src/config/enum.ts

```diff
@@ -7,14 +7,26 @@ export enum Adapter {
   Tidal = 'tidal',
 }
 
+export enum Adapter {
+  Spotify = StreamingService.Spotify,
+  YouTube = StreamingService.YouTube,
+  AppleMusic = StreamingService.AppleMusic,
+  Deezer = StreamingService.Deezer,
+  SoundCloud = StreamingService.SoundCloud,
+  Tidal = StreamingService.Tidal,
+}
+
 export enum Parser {
-  Spotify = 'spotify',
-  YouTube = 'youTube',
-  AppleMusic = 'appleMusic',
-  Deezer = 'deezer',
-  SoundCloud = 'soundCloud',
+  Spotify = StreamingService.Spotify,
+  YouTube = StreamingService.YouTube,
+  AppleMusic = StreamingService.AppleMusic,
+  Deezer = StreamingService.Deezer,
+  SoundCloud = StreamingService.SoundCloud,
+  Tidal = StreamingService.Tidal,
 }
 
+export type StreamingServiceType = Adapter & Parser;
```

Comment:
Since `Parser` and `Adapter` are basically the same `enum`, you could remove one of them to keep one source of truth

Reply:
This would leave out the case: An adapter can exist without a parser and a parser can exist without an adapter (no inverted search supported)